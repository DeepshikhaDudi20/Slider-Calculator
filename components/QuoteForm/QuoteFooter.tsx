import React, { FC } from 'react'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { faAngleDoubleRight } from '@fortawesome/free-solid-svg-icons'
import { QuoteFormStats, QuoteFormStatFields, Value, Label, ButtonContainer, ButtonWrapper, Button } from '../layout/Footer'

interface QuoteFooterProps {
  interest: number;
  repayment: number;
}

const QuoteFooter: FC<QuoteFooterProps> = ({ interest, repayment }) => {

  const onBtnClickHandler = (e: React.MouseEvent<HTMLButtonElement>) => {
    e.preventDefault();
  };

  return (
    <>
      <QuoteFormStats>
        <QuoteFormStatFields>
          <Value data-testid='interest-rate'>{interest}%</Value>
          <Label>Interest rate</Label>
        </QuoteFormStatFields>
        <QuoteFormStatFields>
          <Value data-testid='repayment-amount'>Â£{Math.round(repayment)}</Value>
          <Label>Monthly repayment</Label>
        </QuoteFormStatFields>
      </QuoteFormStats>
      <ButtonContainer>
        <ButtonWrapper>
          <Button data-testid="quote-btn" aria-label='Get your quote' onClick={onBtnClickHandler}>
            Get your quote
            <FontAwesomeIcon icon={faAngleDoubleRight} />
          </Button>
        </ButtonWrapper>
      </ButtonContainer>
    </>
  );
}

export default QuoteFooter;
