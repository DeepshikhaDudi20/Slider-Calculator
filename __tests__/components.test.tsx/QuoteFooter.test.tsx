/**
 * @jest-environment jsdom
 */

import '@testing-library/jest-dom/extend-expect'
import React from 'react';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { axe, toHaveNoViolations } from 'jest-axe';
import QuoteFooter from '../../components/QuoteForm/QuoteFooter';

expect.extend(toHaveNoViolations);

describe('The Quote Footer Component', () => {

  const loanDetails = {
    interest: '5',
    repayment: '80',
  }

  test('should render snapshots', () => {
    const tree = renderer.create(<QuoteFooter />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('should render the loan details on page load', async () => {
    render(<QuoteFooter />);
    const quoteBtnElement = screen.queryByRole('button', { name: /Get your quote/i });
    expect(quoteBtnElement).toBeInTheDocument()
  })

  test("should render button to get quote", async () => {
    render(<QuoteFooter interest={loanDetails.interest} repayment={loanDetails.repayment} />);
    expect(screen.getByTestId('interest-rate').textContent).toEqual('5%')
    expect(screen.getByTestId('repayment-amount').textContent).toEqual('Â£80')
  });

  test("should not have any accessibility violations", async () => {
    const { container } = render(<QuoteFooter interest={loanDetails.interest} repayment={loanDetails.repayment} />);
    expect(await axe(container)).toHaveNoViolations();
  });

});