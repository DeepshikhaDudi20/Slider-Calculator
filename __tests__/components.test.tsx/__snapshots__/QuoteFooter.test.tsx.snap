// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Quote Footer Component should render snapshots 1`] = `
[
  <div
    className="Footer__QuoteFormStats-sc-j8bigy-0 bftBsi"
  >
    <div
      className="Footer__QuoteFormStatFields-sc-j8bigy-1 dtHkZx"
    >
      <span
        className="Footer__Value-sc-j8bigy-2 QZGdB"
        data-testid="interest-rate"
      >
        5
        %
      </span>
      <span
        className="Footer__Label-sc-j8bigy-3 cFZuPt"
      >
        Interest rate
      </span>
    </div>
    <div
      className="Footer__QuoteFormStatFields-sc-j8bigy-1 dtHkZx"
    >
      <span
        className="Footer__Value-sc-j8bigy-2 QZGdB"
        data-testid="repayment-amount"
      >
        Â£
        86
      </span>
      <span
        className="Footer__Label-sc-j8bigy-3 cFZuPt"
      >
        Monthly repayment
      </span>
    </div>
  </div>,
  <div
    className="Footer__ButtonContainer-sc-j8bigy-4 ilBOR"
  >
    <div
      className="Footer__ButtonWrapper-sc-j8bigy-5 bRPHLH"
    >
      <button
        aria-label="Get your quote"
        className="Footer__Button-sc-j8bigy-6 hhoPwJ"
        data-testid="quote-btn"
        onClick={[Function]}
      >
        Get your quote
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-angles-right "
          data-icon="angles-right"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={{}}
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M438.6 278.6c12.5-12.5 12.5-32.8 0-45.3l-160-160c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3L370.7 256 233.4 393.4c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0l160-160zm-352 160l160-160c12.5-12.5 12.5-32.8 0-45.3l-160-160c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3L178.7 256 41.4 393.4c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0z"
            fill="currentColor"
            style={{}}
          />
        </svg>
      </button>
    </div>
  </div>,
]
`;
